frontend/src/app/book-details/book-details.component.ts

```diff
@@ -26,6 +28,36 @@ export class BookDetailsComponent implements OnInit {
       this.bookService.getBookById(id).subscribe(data => {
         this.book = data;
       });
+
+      this.bookService.getBookItemsByBookId(id).subscribe(data => {
+        this.bookItems = data.filter(b => b.status != BookItemStatus.LOST);
+      });
+  }
+
+  getStatusLabel(status: BookItemStatus): string {
+    let st = "";
+    switch(status) {
+      case BookItemStatus.AVAILABLE:
+      case BookItemStatus.RESERVED:
+        st = "Available"; break;
+      case BookItemStatus.LOANED:
+        st = "On loan"; break;
+      default:
+        st = "Unavailable";
+    }
+    return st;
+  }
+
+  getLoanableItems(): BookItem[] {
+    return this.bookItems.filter(item => item.isReferenceOnly === false);
+  }
+
+  getOnSiteItems(): BookItem[] {
+    return this.bookItems.filter(item => item.isReferenceOnly === true)
+  }
+
+  isAvailable(bookItem: BookItem): boolean {
+    return bookItem.status === BookItemStatus.AVAILABLE || bookItem.status === BookItemStatus.RESERVED;
```

Comment:
I tried to create an object method but it did not work. This approach just works.